apps/src/applab/Exporter.js

```diff
@@ -326,11 +324,20 @@ export default {
     const rootRelativeApplabAssetPrefix = rootApplabPrefix + 'assets';
     const zipApplabAssetPrefix = appName + '/' + rootRelativeApplabAssetPrefix;
 
+    // Attempt to fetch applab-api.min.js if possible, but when running on non-production
+    // environments, fallback if we can't fetch that file to use applab-api.js:
+    var applabApiAsset = new $.Deferred();
+    $.when(download('/blockly/js/applab-api.min.js' + cacheBust, 'text'))
+      .then(minAsset => applabApiAsset.resolve([minAsset]), () => {
+        $.when(download('/blockly/js/applab-api.js' + cacheBust, 'text'))
+          .then(asset => applabApiAsset.resolve([asset])), () => applabApiAsset.reject(new Error("failed to fetch applab-api.js"));
+      });
```

Comment:
I'm not super familiar with how close `$.Deferred` is to standard Promises, but I think this would read better with [`Deferred.fail`](https://api.jquery.com/deferred.fail/) callbacks.  Also, doesn't `download` return a thenable `jqXHR`?
```js
download('...', 'text')
  .then(minAsset => applabApiAsset.resolve([minAsset]))
  .catch(() => download('...', 'text'))
  .then(asset => applabApiAsset.resolve([asset]))
  .catch(() => applabApiAsset.reject(new Error("...")));
```

Reply:
It returns 3 params : data, status, jqXHR

Since we only care about the first param below, I only passed that. But we could pass all along...

I’ll try the catch() approach. I was using the pattern below (success and failure callbacks passed to then())

Reply:
`catch()` doesn't seem to work like it would with ES6 promises.
But, I was able to remove the $.when() calls and am now passing all 3 params through.